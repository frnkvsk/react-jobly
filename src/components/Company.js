import React, {useState, useEffect} from 'react';
import Container from '@material-ui/core/Container';
import { makeStyles } from '@material-ui/core/styles';
// import { Button } from '@material-ui/core';
import JobCard from './JobCard';
import JoblyApi from '../api/JoblyApi';


const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    fontSize: '28px',
    maxWidth: 'lg',
    // backgroundColor: '#ffffff',
    // height: '100vh',
    // border: '1px solid red'
  },
  header: {
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
    margin: '20px 0 20px 20px',
    paddingTop: '30px',

  },
  card: {
    display: 'flex',
    justifyContent: 'space-between',
    width: '100%',
    padding: '20px 0 20px 20px',
    margin: '10px 0 10px 0',
    border: '1px solid lightgray',
  },
  btn: {
    // display: 'absolute',
    // position: 'relative',
    // top: '50px',
    // left: '30px',
    margin: '60px 30px 0 0',
    height: '30px', 
  }
}));
// const getComp = async comp => {
//   return await JoblyApi.getCompany(comp);
// }
const Company = ({handle}) => {
  const classes = useStyles();
  const [company, setCompany] = useState({});
  const [jobs, setJobs] = useState([]);
  useEffect(() => {
    const getValue = async () => {
      const value = await JoblyApi.getCompany(handle);
      setCompany(value);
    }
    getValue();
  });
  useEffect(() => {
    const getValue = async () => {
      const value = await JoblyApi.getJobs();;
      setJobs(value.filter(e => e.company_handle === handle));
    }
    getValue();
  });
  return (
    <>
      <Container className={classes.root} >
        <div className={classes.header}>
          <div><b>{company.name}</b></div>
          <div>{company.description}</div>
        </div>
        {jobs.map(job => (
          <JobCard job={job} />
        ))}
      </Container>
    </>
  );  
}

// const useGetCompany = handle => {
//   const [value, setValue] = useState({});
//   useEffect(() => {
//     const getValue = async () => {
//       const value = await JoblyApi.getCompany(handle);
//       setValue(value);
//     }
//     getValue();
//   });
//   return value;
// }
// const useGetJobs = handle => {
//   const [value, setValue] = useState([]);
//   useEffect(() => {
//     const getValue = async () => {
//       const value = await JoblyApi.getJobs();;
//       setValue(value.filter(e => e.company_handle === handle));
//     }
//     getValue();
//   });
//   return value;
// }
export default Company;