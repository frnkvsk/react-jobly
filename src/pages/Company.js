import React, { useState, useEffect, useContext } from 'react';
import { useParams } from 'react-router-dom';
import Container from '@material-ui/core/Container';
import { makeStyles } from '@material-ui/core/styles';
import JobCard from '../components/JobCard';
import { getJobs, getCompany } from '../api/JoblyApi';
import { AuthContext } from '../context/AuthContext';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'flex-start',
    minHeight: '100vh',
    fontSize: '28px',
    maxWidth: 'lg',
  },
  header: {
    display: 'flex',
    flexDirection: 'column',
    width: '100%',
    margin: '20px 0 20px 20px',
    paddingTop: '30px',
  },
  card: {
    display: 'flex',
    justifyContent: 'space-between',
    width: '100%',
    padding: '20px 0 20px 20px',
    margin: '10px 0 10px 0',
    border: '1px solid lightgray',
  },
  btn: {
    margin: '60px 30px 0 0',
    height: '30px', 
  }
}));

const Company = () => {
  const classes = useStyles();
  const {handle} = useParams();
  const [company, setCompany] = useState(null);
  const [jobs, setJobs] = useState([]);
  const auth = useContext(AuthContext);
  const params = {_token: auth.authState.token};
  useEffect(() => {
    const getValue = async () => {
      const value = await getCompany(params, handle);
      if(value) {
        setCompany(value.company);
      }      
    }
    getValue();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  useEffect(() => {
    const getValue = async () => {
      const value = await getJobs(params);
      if(value) {
        setJobs(value.jobs.filter(e => e.company_handle === handle));
      }      
    }
    if(!jobs.length) getValue();
  });
  return (
    <>
      <Container className={classes.root} >
        {
          company ? 
            <div className={classes.header}>
              <div><b>{company.name}</b></div>
              <div>{company.description}</div>
            </div> :
            <h1>...Loading</h1>
        }
        
        {
          jobs.length ? jobs.map(job => (
            <JobCard key={job.id} job={job} />
          )) : 
          <h1>...Loading</h1>
        }
      </Container>
    </>
  );  
}

export default Company;