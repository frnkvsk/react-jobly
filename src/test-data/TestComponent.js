// import React, { useContext } from 'react';
// import { AuthContext } from '../context/AuthContext';

// const TestComponent = () => {
//   const auth = useContext(AuthContext);
//   const setAuth2 = async () =>
//     await auth.setAuthState({
//       token: 'testtoken',
//       userInfo: {
//       username: 'testuser',
//       first_name: 'testfirst',
//       last_name: 'testlast',
//       email: 'test@test.com',
//       photo_url: null
//     }});
//   (async () => await setAuth2())();
//   console.log('-------auth.authState',auth.authState)
//   return (
//     <div>
//       <div>{auth.authState.userInfo.username}</div>
//       <div>{auth.authState.userInfo.first_name}</div>
//       <div>{auth.authState.userInfo.last_name}</div>
//       <div>{auth.authState.userInfo.email}</div>
//       <div>{auth.authState.userInfo.photo_url}</div>
//     </div>
//   )
// }
// export default TestComponent;

