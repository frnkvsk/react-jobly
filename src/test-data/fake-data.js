const testData = {
  jobs: [
    {
    company_handle: "test-company-a",
    equity: 0.115,
    id: 1,
    salary: 11111,
    state: null,
    title: "Editor, magazine features",
    },
      {
      company_handle: "test-company-b",
      equity: 0.22,
      id: 2,
      salary: 22222,
      state: null,
      title: "Coder",
    },
      {
      company_handle: "test-company-c",
      equity: 0.33,
      id: 3,
      salary: 33333,
      state: null,
      title: "Front-end Dev",
    },
  ],
  companies: [
    {
      description: "Test description 1",
      handle: "test-company-1",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 1",
    },
    {
      description: "Test description 2",
      handle: "test-company-2",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 2",
    },
    {
      description: "Test description 3",
      handle: "test-company-3",
      logo_url: "http://www.gtdesigns.it/wp-content/uploads/OverusedLogos/99gen_circle.jpg",
      name: "Test Company 3",
    },
  ],
}

export { testData }