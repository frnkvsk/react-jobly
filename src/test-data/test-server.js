import { rest } from 'msw';
import { setupServer } from 'msw/node';

let server;

const setServer = async (url, status, message='') => {
  if(server) {
    server.close();
    server.resetHandlers();
  }
  server = await setupServer(
    rest.get(url, (req, res, ctx) => {
      return res(
        ctx.status(status),
        ctx.json(message)
      )
    })
  );
}
  

beforeAll(() => server ? server.listen() : null);
afterAll(() => server ? server.close() : null);
afterEach(() => server ? server.resetHandlers() : null);

export { setServer, server, rest }
