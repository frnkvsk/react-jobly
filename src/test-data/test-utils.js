// test-utils.js
import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';
import { SearchProvider } from '../context/SearchContext';
import { AuthProvider } from '../context/AuthContext';
import { PageCountProvider } from '../context/PageCountContext';

const AllTheProviders = (page) => {
  return render(
    <BrowserRouter>
      <AuthProvider>
        <SearchProvider>
          <PageCountProvider>
            {page}
          </PageCountProvider>
        </SearchProvider>
      </AuthProvider> 
    </BrowserRouter>
    
  );
}

// re-export everything
export * from '@testing-library/react';

// override render method
export { AllTheProviders as render }
