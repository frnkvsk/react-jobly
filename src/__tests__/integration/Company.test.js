import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '../../test-data/test-utils';
import Company from '../../pages/Company';

// smoke and appearance tests
test('renders Company ', () => {
  render(<Company />);
  // expect(screen.getAllByText(/...Loading/i)[0]).toBeInTheDocument();
});

// snapshot test
// test('matches snapshot', () => {
//   const { asFragment } = render(<Company />);
//   expect(asFragment()).toMatchSnapshot();
// });