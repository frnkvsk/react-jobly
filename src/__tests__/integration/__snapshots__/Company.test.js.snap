// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root makeStyles-root-5 MuiContainer-maxWidthLg"
  >
    <h1>
      ...Loading
    </h1>
    <h1>
      ...Loading
    </h1>
  </div>
</DocumentFragment>
`;
